pkg/rhoda/rhoda_test.go

```diff
@@ -18,165 +22,379 @@ import (
-
+			if err != nil {
+				AbortSuite(err.Error())
+			}
 			Expect(err).NotTo(HaveOccurred())
 		})
 	})
 
-	Context("Populate providers array", func() {
-		It("Get list of providers from the vault, create secrets, populate array", func() {
-			//Get ci-secret's data
-			clientset, err := kubernetes.NewForConfig(config)
-			Expect(err).NotTo(HaveOccurred())
-
-			ciSecret, err := clientset.CoreV1().Secrets("osde2e-ci-secrets").Get(context.TODO(), "ci-secrets", meta.GetOptions{})
-			Expect(err).NotTo(HaveOccurred())
+	Context("Get all the providers and loop through it to create Secrets and Inventory", func() {
+		var providerNames []string
+		var providers []rhoda.ProviderAccount
+		//Get ci-secret's data
+		clientset, err := kubernetes.NewForConfig(config)
+		Expect(err).NotTo(HaveOccurred())
+		ciSecret, err := clientset.CoreV1().Secrets("osde2e-ci-secrets").Get(context.TODO(), "ci-secrets", meta.GetOptions{})
+		if err != nil {
+			AbortSuite(err.Error())
+		}
+		fmt.Println("ciSecret Found: ")
 
-			fmt.Println("ciSecret Found: ")
-			//get the list of providers by getting providerList secret
-			var providerNames []string
-			if providerListSecret, ok := ciSecret.Data["providerList"]; ok {
-				fmt.Printf("providerListSecret = %s, ok = %v\n", providerListSecret, ok)
-				providerNames = strings.Split(string(providerListSecret), ",")
-				fmt.Println(providerNames)
+		//get the list of providers by getting providerList secret
+		if providerListSecret, ok := ciSecret.Data["providerList"]; ok {
+			//fmt.Printf("providerListSecret = %s, ok = %v\n", providerListSecret, ok)
+			providerNames = strings.Split(string(providerListSecret), ",")
+			fmt.Println(providerNames)
+		} else {
+			AbortSuite("ProviderList secret was not found")
+		}
+		//populate providers array
+		for _, providerName := range providerNames {
+			fmt.Println(providerName)
+			var secretData = make(map[string][]byte)
+			for key, value := range ciSecret.Data {
+				if strings.HasPrefix(key, providerName) {
+					fmt.Printf("    %s: %s\n", key, value)
+					var keyName = strings.Split(key, "-")
+					//create map of secret data
+					secretData[keyName[1]] = value
+				}
 			}
+			//add provider's data to an array
+			providers = append(providers, rhoda.ProviderAccount{ProviderName: providerName, SecretName: "dbaas-secret-e2e-" + providerName, SecretData: secretData})
+		}
+
+		//add dbaas scheme for inventory creation
+		scheme := runtime.NewScheme()
+		err = dbaasv1alpha1.AddToScheme(scheme)
+		Expect(err).NotTo(HaveOccurred())
+
+		client, err := k8sClient.New(config, k8sClient.Options{Scheme: scheme})
+		Expect(err).NotTo(HaveOccurred())
+
+		//loop through providers
+		for i := range providers {
+			provider := providers[i]
+			It("Should pass when secret and provider created and connection status is checked for "+provider.ProviderName, func() {
+				DeferCleanup(func() {
+					fmt.Println("DeferCleanup Started")
```

Comment:
I think we should simplify and shorten this cleanup. We are no longer testing, and no need to check and fail. Just delete and move on.

Reply:
what if it did not delete?

Reply:
It doesn't matter... we're not actually testing k8s deletes, we're just cleaning up. If we can't clean up, then we can't. No reason to check.